config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml

```diff
@@ -0,0 +1,132 @@
+periodics:
+# Daily run, to enforce the state of the manifest on a regular basis. This could
+# happen more frequently (perhaps hourly), but it's probably not a big deal,
+# given how the postsubmit will take care of all needs (in theory). This is more
+# of a sanity check in cases where there are no merges to "master" branch for
+# long periods of time.
+- interval: 24h
+  max_concurrency: 1
+  agent: kubernetes
+  # This name is the "job name", passed in as "--job=NAME" for mkpj.
+  name: cip-sanity
+  labels:
+    preset-mount-service-account-key: "true"
+  # Enable Pod Utilities. See https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md.
+  decorate: true
+  extra_refs:
+  # Because of Pod Utilities, we clone the below repo automatically, and, get
+  # dropped into /home/prow/go/src/github.com/kubernetes/k8s.io.
+  - org: kubernetes
+    repo: k8s.io
+    base_ref: master
+  spec:
+    containers:
+    # TODO: Move the official cip image to a more serious location. But it's not
+    # a huge problem because we pin it by the digest.
+    - image: gcr.io/cip-demo-staging/cip@sha256:3823b84b54b2e099f4b6e51e4a6225f04c1bf10c606313fb85a3cfff4df8d318
+      args:
+      - /bin/sh
+      - -c
+      - |
+        # This is aided by Pod Utilities because of "decorate: true". , which already has the
+        # manifest for us. So there's no need to clone, as in the periodics.
+        # There is also PULL_REFS and PULL_BASE_SHA defined for us.
+        #
+        # If you run mkpj for this postsubmit, the job will query you for the
+        # base ref to fetch (name of a branch, not a SHA, e.g. "master").
+        k8s.gcr.io/run-cip.sh \
+          /app/cip-docker-image.binary \
+          k8s-staging-cluster-api/manifest.yaml,/etc/cip-service-account/service-account.json \
+          k8s-staging-coredns/manifest.yaml,/etc/cip-service-account/service-account.json \
+          k8s-staging-csi/manifest.yaml,/etc/cip-service-account/service-account.json
+      # TODO: Uncomment this once
+      # https://github.com/GoogleCloudPlatform/k8s-container-image-promoter/issues/16
+      # lands.
+      #env:
+      #- name: CIP_OPTS
+      #  # Be aggressive about keeping the prod registry clean.
+      #  value: "-delete-extra-tags"
+      imagePullPolicy: Always
+
+presubmits:
+  kubernetes/k8s.io:
+  - name: cip-dryrun
+    decorate: true
+    labels:
+      preset-mount-service-account-key: "true"
+    skip_report: false
+    run_if_changed: ".*/manifest.yaml"
+    # It's OK to run things concurrently, because we're only running things with -dry-run.
+    max_concurrency: 10
+    branches:
+    - ^master$
+    spec:
+      containers:
+      - image: gcr.io/cip-demo-staging/cip@sha256:3823b84b54b2e099f4b6e51e4a6225f04c1bf10c606313fb85a3cfff4df8d318
+        args:
+        - /bin/sh
```

Comment:
Will you please encapsulate this logic into a `something.sh` script in the kubernetes/k8s.io repo? 

Note that this is how prow does things:
https://github.com/kubernetes/test-infra/blob/1ab0846d5d59ec1ace2a548983164ae7eacc088c/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L59-L60

The logic for this we then check into the repo: https://github.com/kubernetes/test-infra/blob/master/prow/push.sh

Reply:
FWIW the logic was encapsulated like this already: https://github.com/kubernetes/k8s.io/pull/203/files

The reason I didn't encapsulate _all_ of the logic (including the arguments such as `k8s-staging-cluster-api/manifest.yaml,/etc/cip-service-account/service-account.json`) is because the logic that determines the location of the service account creds file lives inside the configuration of this prow job itself.

That being said, are you still opposed to calling the script with arguments like this?

Reply:
The thing I don't like is the `/bin/sh`. Is there a reason you cannot use:
```yaml
command:
- multirun.sh
args:
- /app/cip-docker-image.binary
- k8s-staging-cluster-api/manifest.yaml
- ...
```